backend/associations/views/election.py

```diff
@@ -1,92 +1,220 @@
+                "The status cannot be updated as the election has not started."
+            )
+
+        # The status can only be updated from PENDING.
+        if self.get_object().status != "PENDING":
+            raise ValidationError("The status cannot be updated anymore.")
+
+        # The status can only be updated to OFFLINE_VOTE (as ONLINE_VOTE is automatically handled).
+        if serializer.validated_data["status"] != "OFFLINE_VOTE":
+            raise ValidationError("The status can only be updated to OFFLINE_VOTE.")
+
+        # Do not change the user and the election.
+        serializer.save(
+            user=self.get_object().user, election=self.get_object().election
+        )
+
+    def perform_destroy(self, instance: Voter):
+        # Only allow deletion if the election has not started.
+        if instance.election.started:
+            raise ValidationError("The voter cannot be deleted anymore.")
+
+        instance.delete()
+
+
+class ChoiceViewSet(viewsets.ModelViewSet):
+    queryset = Choice.objects.all()
+    serializer_class = ChoiceSerializer
+    permission_classes = (ChoicePermission,)
+
+    filter_fields = ("election", "election__association")
+
+    def get_queryset(self):
+        """Filter the choices according to the user election permission."""
+        return Choice.objects.filter(
+            election__association__id__in=self.request.user.get_associations_with_permission(
+                "election"
+            )
+        )
+
+    def perform_create(self, serializer):
+        # Check if the user can create the choice.
+        role = self.request.user.get_role(
+            serializer.validated_data["election"].association
+        )
+        if not role or not role.election:
+            raise PermissionDenied(
+                "You are not allowed to create a choice for this election."
+            )
+
+        # Only allow creation if the election has not started.
+        if serializer.validated_data["election"].started:
+            raise ValidationError("The choice cannot be added anymore.")
 
-"""
-    Endpoints:
-        * List elections:               GET     /associations/elections/
-        * Retrieve an election:         GET     /associations/elections/1/
-        * Create an election:           POST    /associations/elections/1/
-        * Update an election:           PATCH   /associations/elections/1/
-        * Destroy an election:          DELETE  /associations/elections/1/
-        * Vote to an election:          POST    /associations/elections/1/vote/
-        * Results of an election:       GET     /associations/elections/1/results/
-"""
+        serializer.save()
+
+    def perform_update(self, serializer):
+        choice = self.get_object()
+        override = {"election": choice.election}
+
+        # If the election has started, do not change the name.
+        if self.get_object().election.started:
+            override["name"] = choice.name
```

Comment:
else send 403 ou 400

Reply:
You mean if the election has not started?

Reply:
But if the election has not started we can still edit